geode-connectors/src/distributedTest/java/org/apache/geode/connectors/jdbc/internal/cli/CreateMappingCommandDUnitTest.java

```diff
@@ -61,12 +72,95 @@ public void before() throws Exception {
 
     gfsh.connectAndVerify(locator);
 
-    gfsh.executeAndAssertThat("create region --name=" + REGION_NAME + " --type=REPLICATE")
+  }
+
+  private void setupReplicate() {
+    setupReplicate(false);
+  }
+
+  private void setupReplicate(boolean addLoader) {
+    gfsh.executeAndAssertThat("create region --name=" + REGION_NAME + " --type=REPLICATE"
+        + (addLoader ? " --cache-loader=" + JdbcLoader.class.getName() : ""))
         .statusIsSuccess();
   }
 
+  private void setupPartition() {
+    gfsh.executeAndAssertThat("create region --name=" + REGION_NAME + " --type=PARTITION")
+        .statusIsSuccess();
+  }
+
+  private void setupAsyncEventQueue() {
+    gfsh.executeAndAssertThat(
+        "create async-event-queue --id=" + CreateMappingCommand.getAsyncEventQueueName(REGION_NAME)
+            + " --listener=" + JdbcAsyncWriter.class.getName())
+        .statusIsSuccess();
+  }
+
+  private static RegionMapping getRegionMappingFromClusterConfig() {
+    CacheConfig cacheConfig =
+        InternalLocator.getLocator().getConfigurationPersistenceService().getCacheConfig(null);
+    RegionConfig regionConfig = cacheConfig.getRegions().stream()
+        .filter(region -> region.getName().equals(REGION_NAME)).findFirst().orElse(null);
+    return (RegionMapping) regionConfig.getCustomRegionElements().stream()
+        .filter(element -> element instanceof RegionMapping).findFirst().orElse(null);
+  }
+
+  private static RegionMapping getRegionMappingFromService() {
+    return ClusterStartupRule.getCache().getService(JdbcConnectorService.class)
```

Comment:
We still call this extension "the JDBC Connector". This is the "service" for "the JDBC Connector".
We still use it to keep track of our jdbc mappings in memory. In the future, as features are added to the jdbc connector it may also keep track of other things. So I think we should keep it.
This is the "best practice" for each extension. The service feature allows the extension to have state it can find off the cache without adding new apis to the cache (like we used to do with things like the FunctionService and the QueryService).